src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -1630,6 +1638,14 @@ void MainWindow::slot_start_rendering_once(const QString& filepath, const QStrin
     }
 }
 
+void MainWindow::slot_pause_or_resume_rendering()
+{
+    if (m_rendering_manager.is_rendering() && !m_rendering_manager.is_rendering_paused())
```

Comment:
I need an advice for this. I suppose the slot will be called only while rendering (otherwise the widget is disabled). So the first condition `is_rendering` can be removed. What do you think ?

Reply:
I think so, yes. You can check `is_rendering()` in an assert.